connpicker.go

```diff
@@ -0,0 +1,87 @@
+package gocql
+
+import (
+	"sync"
+	"sync/atomic"
+)
+
+type ConnPicker interface {
+	Pick(token) *Conn
+	Put(*Conn)
+	Remove(conn *Conn)
+	Size() (int, int)
+	Close()
+}
+
+type DefaultConnPicker struct {
+	conns []*Conn
+	pos   uint32
+	size  int
+	mu    sync.RWMutex
+}
+
+func NewDefaultConnPicker(size int) *DefaultConnPicker {
+	return &DefaultConnPicker{
```

Comment:
check for size to be non-zero to avoid errors?

Reply:
I can check it here and return an error. If so what do we do then, panic or default to something?

Reply:
There is no real infrastructure available to handle errors in this. It all relies on the config defaulting to 2. We can change the assumptions but I'd rather not in a patch that is about something else.

What do you think?

Reply:
I think panic here is fine.